src/agent/src/rpc.rs

```diff
@@ -1737,6 +1774,137 @@ fn load_kernel_module(module: &protocols::agent::KernelModule) -> Result<()> {
     }
 }
 
+fn pull_image_from_registry(
+    image: &str,
+    cid: &str,
+    source_creds: &Option<&str>,
+    use_aa: bool,
+    agent_started: bool,
+) -> Result<()> {
+    let source_image = format!("{}{}", "docker://", image);
+
+    let manifest_path = format!("/tmp/{}/image_manifest", cid);
+    let target_path_manifest = format!("dir://{}", manifest_path);
+
+    // Define the target transport and path for the OCI image, without signature
+    let oci_path = format!("/tmp/{}/image_oci:latest", cid);
```

Comment:
- Can you make `image_oci:latest` a `const` as it's hard-coded multiple times?
- Would there ever be a need to *not* use latest? We often see problems with the latest versions so it might be prudent to allow the specification of an alternative image somehow.

Reply:
Sure, this is not part of my patch tho. I guess the rebase has kind of confounded this PR.

Reply:
Ack. As mentioned on another #2658, imho we should try to get these CC PRs landed into the `CCv0` branch as soon as we can, close outstanding stale PRs and potentially break the PRs into smaller units to speed up the process. I understand how we've got to the current situation, but it's starting to get extremely confusing, atleast to me.